docs/topics/proc_exposing-application-metrics-using-prometheus-with-vertx.adoc

```diff
@@ -105,54 +108,43 @@ import io.vertx.micrometer.backends.BackendRegistries;
 
 public class MyVertxApp extends AbstractVerticle {
 
+  public class CustomLauncher extends Launcher {
+
+    @Override
+    public void beforeStartingVertx(VertxOptions options) {
+      options.setMetricsOptions(new MicrometerMetricsOptions()
+        .setPrometheusOptions(new VertxPrometheusOptions().setEnabled(true)
+          .setStartEmbeddedServer(true)
+          .setEmbeddedServerOptions(new HttpServerOptions().setPort(8081))
+          .setEmbeddedServerEndpoint("/metrics"))
+        .setEnabled(true));
+    }
+
     @Override
-    public void start() {
```

Comment:
You must keep a `start` method with some code inside, otherwise you're not monitoring anything:

```java
@Override
public void start() {
  MeterRegistry registry = BackendRegistries.getDefaultNow();
  Timer timer = Timer
    .builder("my.timer")
    .description("a description of what this timer does")
    .register(registry);

  vertx.setPeriodic(1000, l -> {
    timer.record(() -> {
      // Do something
    });
  });
}
```

Reply:
Where exactly does this code block appear? Right after line 109?

Reply:
It should be the only method in the verticle class